ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java

```diff
@@ -24,18 +24,62 @@
 import io.confluent.ksql.query.QueryId;
 import io.confluent.ksql.query.id.QueryIdGenerator;
 import io.confluent.ksql.util.KsqlException;
+
+import java.util.Arrays;
+import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.ThreadLocalRandom;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 /**
  * Utility for constructing {@link QueryId}s - separate from {@code EngineExecutor} for
  * easy access to unit testing.
  */
-final class QueryIdUtil {
+public final class QueryIdUtil {
+  private static final Pattern VALID_QUERY_ID = Pattern.compile("[A-Za-z0-9_]+");
+  private static final ReservedQueryIdsPrefixes[] PREFIXES = ReservedQueryIdsPrefixes.values();
+
+  public enum ReservedQueryIdsPrefixes {
+    INSERT("INSERTQUERY_"),
+    CTAS("CTAS_"),
+    CSAS("CSAS_");
+
+    private final String prefix;
+    ReservedQueryIdsPrefixes(final String prefix) {
+      this.prefix = prefix;
+    }
+
+    @Override
+    public String toString() {
+      return prefix;
+    }
+  }
 
   private QueryIdUtil() {
   }
 
+  private static void validateWithQueryId(final String queryId) {
```

Comment:
we should make this a `ConfigDef.Validator` and validate it as part of the config def

Reply:
I initially thought of this, but I decided to do the validation in the `QueryIdUtil` to make sure that whoever wants to use that class, then they pass the right QueryID string. I don't think nobody else will use it, but I think is a better design as the `buildId` is the one will check IDs are correct.

Unless you have another reason to use the Validator?

Reply:
yeah that makes sense